src/core/include/openvino/core/layout.hpp

```diff
@@ -88,7 +90,7 @@ class OPENVINO_API Layout {
     friend Layout layout::apply_permutation(const Layout& src_layout, const std::vector<uint64_t>& dims);
 
     friend std::vector<int64_t> layout::find_permutation(const Layout& src_layout,
-                                                         const Rank& src_shape_rank,
+                                                         const PartialShape& src_shape,
                                                          const Layout& dst_layout);
 };
```

Comment:
BTW, can we move these functions to `src`? They are not exported, but still usable by users of static OpenVINO libraries

Reply:
Done. But to hide function signatures from 'friend declaration' I had to move implementation into 'class LayoutUtils'